lib/int64.js

```diff
@@ -0,0 +1,355 @@
+'use strict';
+
+const INT32_MAX = 0x7FFFFFFF;
+const UINT32_MAX = 0xFFFFFFFF;
+
+class Int64 {
+  constructor(value) {
+    this.value = new Uint32Array(2);
+    if (value instanceof Int64) {
+      this.value[0] = value.value[0];
+      this.value[1] = value.value[1];
+      return;
+    }
+    const numValue = Number(value);
+    if (Number.isNaN(numValue)) return;
+    if (numValue >= 0 && numValue <= INT32_MAX) {
+      this.value[0] = numValue;
+    } else if (numValue >= -INT32_MAX && numValue < 0) {
+      this.value[0] = numValue;
+      this.value[1] = UINT32_MAX;
+    } else if (numValue >= 0 && numValue <= Number.MAX_SAFE_INTEGER) {
+      this.value[0] = numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(numValue / (UINT32_MAX + 1));
+    } else if (numValue < 0 && numValue >= Number.MIN_SAFE_INTEGER) {
+      this.value[0] = -numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(-numValue / (UINT32_MAX + 1));
+      this.value = Int64._conversion(this.value);
+    } else if (typeof value === 'string') {
+      if (value.startsWith('0x')) {
+        value = value.slice(2);
+        this.value[0] = Number('0x' + value.slice(value.length - 8));
+        this.value[1] = Number('0x' + value.slice(0, value.length - 8));
+      } else if (value.startsWith('0o')) {
+        value = value.slice(2);
+        const val0 = Number('0o' + value.slice(value.length - 11));
+        const val1 = Number('0o' + value.slice(0, value.length - 11));
+        this.value[0] = val0;
+        this.value[1] = (val1 << 1) | (val0 >>> 31);
+      } else if (value.startsWith('0b')) {
+        value = value.slice(2);
+        this.value[0] = Number('0b' + value.slice(value.length - 32));
+        this.value[1] = Number('0b' + value.slice(0, value.length - 32));
+      } else {
```

Comment:
Those conversions won't work if the string  is longer than supported by Number. I'd suggest to just use the way it was done in Uint64.